### PR TITLE
Improve Distributed Quiz Exercise Cache Code

### PR DESCRIPTION
**Pull request by MaisiKoleni**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context
Fix some code style issues and improve the readability, add more documentation, explain the reasoning and make the code easier understandable and maintainable.

### Description
- Removed some unnecessary explicit annotation attributes 
- Used `null` as id for the empty quiz exercise cache instead of `-1`
- Renamed some attributes, methods and classes
- **Encapsulated the quiz exercise caches and introduced a new class**
- Throw an `UnsupportedOperationException` when `clear()` is called on the empty cache.
- Added a *lot more* JavaDoc to classes, methods, fields and statements.
- Added and modified integration tests

Only the `clear()` and `null`-id changes should change functionality (that should never be used anywhere).

### Steps for Testing
- **Most important: review the code and check that it is possible to understand what it does.** Check if everything looks right.
- Execute (quiz) tests
- Test participating in Artemis Quiz Exercises, check that no submission is lost.

### CODE REVIEW

**Comment by TobiasPr** [comment]
Tested "conduction of quizzes" together with @fde312 on test server 1 and it worked. I did not review the code / documentation, because I am unfamiliar with the distributed quiz scheduling and it would take much time to understand in detail what is going on. Maybe someone who knows already about Hazelcast and how it works is more suited to judge the comments

**Review by fde312 — APPROVED** [review]
I tested it on the test sever with @TobiasPr and everything worked fined. 
Went through the code and I could not find anything missing in the encapsulation. Good work 🚀

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/quiz/QuizExerciseDistributedCache.java [hunk 1]
```diff
@@ -34,15 +41,25 @@
 
     private static final String HAZELCAST_CACHE_RESULTS = "-results";
 
+    /**
+     * All {@link List} classes that are supported by Hazelcast {@link SerializationServiceV1}
```
Comment (sleiss) [inline src/main/java/de/tum/in/www1/artemis/service/scheduled/quiz/QuizExerciseDistributedCache.java:45]:
```suggestion
     * All {@link List} of classes that are supported by Hazelcast {@link SerializationServiceV1}
```

**Review by sleiss — APPROVED** [review]
Documentation & Tests look good to me, did however not test as @fde312 and @TobiasPr already tested it.

**Review by sleiss — APPROVED** [review]

**Review by jpbernius — APPROVED** [review]

**Pull request merged**